app/views/components/check_uk_visa/_result-card.html.erb

```diff
@@ -0,0 +1,60 @@
+<%
+  type ||= nil
+  title ||= nil
+  description ||= nil
+  attributes ||= []
+  url ||= nil
+
+  if !type || !title || !url || !attributes.any? || !calculator
```

Comment:
Is the [results card validation](https://github.com/alphagov/smart-answers/pull/6141/commits/e90f88ee135ba8aed644f542c58a8fa94412ce95#diff-bb4685394a5e3a9f03f288b3a808404314e69033c7e806394f9de77aae8eb6c6R8) to ensure the YAML data is valid? If so, I think we should probably convert this validation into a test, rather than at runtime validation as the goal is to test our own config, not users input. Errors could also slip through to production if our checks / test coverage isn't comprehensive.

I'd suggest adding a test that parses the YAML file and checks the data integrity within. The [calculator spec](https://github.com/alphagov/smart-answers/blob/main/test/unit/calculators/uk_visa_calculator_test.rb) probably makes most sense, or potentially the [result card spec](https://github.com/alphagov/smart-answers/blob/91eeede53e1cc80f2bfc35e1e521304ae425b0ad/test/components/check_uk_visa/result_card_test.rb.

Thoughts?

Reply:
This sounds good to me, although it seems we have a pattern here based on the other components in
https://github.com/alphagov/smart-answers/tree/91eeede53e1cc80f2bfc35e1e521304ae425b0ad/app/views/components

Reply:
Me and Ed discussed the above offline and agreed that tests would be more suited, rather than view validation for this use case. As this is an emerging pattern and affects other components, it's deemed as non-blocking. However, we have created a tech-debt card regarding this issue: https://trello.com/c/47h58IB7/1609-convert-view-level-validation-pattern-to-equivalent-tests-for-components-in-smart-answers.